### PR TITLE
[UserBackgrounds] Guild Banner Suggestion

### PR DESCRIPTION
**Pull request by HypedDomi**
Hope I did it right. The code is to complex for me

### CODE REVIEW

**Comment by HypedDomi** [comment]
For the Guild Banner i need the Guild Banner hash. Besides that it should work

**Comment by HypedDomi** [comment]
Everything should now work correctly and shouldn't crash. But it would be good if you look over it to be sure

File: UserBackgrounds/index.tsx [hunk 1]
```diff
@@ -119,12 +134,13 @@ export default class UserBackgrounds extends BasePlugin {
...
-        function BannerContainer({user, bannerType, children}: params) {
+        function BannerContainer({user, guildId, bannerType, children}: params) {
+            const guildMember = DiscordModules.GuildMemberStore.getMember(guildId, user.id);
             const banner: banner = useStateFromStores([BannerStore], () => BannerStore.getBanner(user.id), null, _.isEqual);
-            const [selection, setSelection] = useState(banner == null ? 1 : 0);
+            const [selection, setSelection] = useState(guildMember?.banner == null ? user.banner == null ? 2 : 1 : 0 );
             const ref = useRef(null);
-            const currentBanner = useMemo(() => (selection === 1 || banner == null) ? getBannerURL(user, true) : banner?.background, [banner, user, selection]);
-            const currentOrientation = useMemo(() => (banner != null && selection === 0) ? banner.orientation : void 0, [banner, selection]);
+            const currentBanner = useMemo(() => ((selection === 0 || banner == null) && guildMember?.banner) ? getGuildBannerURL(guildMember, guildId, true) : (selection === 1 || banner == null) ? getBannerURL(user, true) : banner?.background, [banner, user, selection]);
```
Comment (Strencher) [inline UserBackgrounds/index.tsx:142]: This could use some cleanup and less ternaries.

File: UserBackgrounds/index.tsx [hunk 2]
```diff
@@ -119,12 +134,13 @@ export default class UserBackgrounds extends BasePlugin {
             children: React.ReactElement;
         };
 
-        function BannerContainer({user, bannerType, children}: params) {
+        function BannerContainer({user, guildId, bannerType, children}: params) {
+            const guildMember = DiscordModules.GuildMemberStore.getMember(guildId, user.id);
```
Comment (Strencher) [inline UserBackgrounds/index.tsx:138]: Please connect it to the store by using
```js
const guildMember = useStateFromStores([DiscordModules.GuildMemberStore], () => 
	DiscordModules.GuildMemberStore.getMember(guildId, user.id)
);
```

File: UserBackgrounds/index.tsx [hunk 3]
```diff
@@ -81,6 +81,21 @@ const getBannerURL = function (user: User, animated = false) {
     }
 };
 
+const getGuildBannerURL = function (user: User, guildId, animated = false) {
```
Comment (Strencher) [inline UserBackgrounds/index.tsx:84]: While you're at, assign type info to `guildId`, so I'd be `guildId: string`

File: UserBackgrounds/index.tsx [hunk 4]
```diff
@@ -176,15 +192,15 @@ export default class UserBackgrounds extends BasePlugin {
...
+                    <TextBadge color={Colors.BRAND_NEW_500} text={selection === 0 ? "GUILD" : selection === 1 ? "NATIVE" : "USRBG"} className={styles.badge} />
                     {
                         (banner != null && user.banner != null) && (
                             <Button
                                 className={Utilities.className(styles.arrow, styles.left)}
                                 key="left"
                                 look={Button.Looks.BLANK}
                                 size={Button.Sizes.TINY}
-                                onClick={setSelection.bind(null, 0)}
+                                onClick={selection === 1 ? setSelection.bind(null, 0) : setSelection.bind(null, 1)}
```
Comment (Strencher) [inline UserBackgrounds/index.tsx:203]: Better practice would be to do something like this instead: (same for the usage below)
```js
setSelection.bind(null, selection === 1 ? 0 : 1)
```

**Review by Strencher — CHANGES_REQUESTED** [review]

File: UserBackgrounds/index.tsx (hunk 1 above)
Comment (HypedDomi) [reply UserBackgrounds/index.tsx]: Tbh, idk how i could clean it up. maybe because its late or im just dumb

Comment (Strencher) [reply UserBackgrounds/index.tsx]: Convert it to use normal `if() {} else {}` structure.

**Comment by HypedDomi** [comment]
When everything is fine, i would bump the version

**Review by Strencher — APPROVED** [review]

**Pull request merged**